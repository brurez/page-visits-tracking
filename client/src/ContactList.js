import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

import LoadingMsg from './LoadingMsg';
import axios from 'axios/index';

class ContactList extends Component {
  state = { data: [], isFetching: true };

  componentDidMount() {
    axios
      .get('/api/contacts')
      .then(res => {
        //debugger;
        const { data, count } = res.data;
        this.setState({ data, isFetching: false });
      })
      .catch(err => console.log(err));
  }

  renderContacts() {
    const { data, isFetching } = this.state;

    return data.map(contact => {
      const { name, email, visits, createdAt } = contact;
      //debugger;
      return (
        <div className="card" key={contact._id}>
          <div className="content">
            <div className="header">{name}</div>
            <div className="meta">
              Criado em: {moment(createdAt).format('LLL')}
            </div>
            <div className="description">
              <strong>Email:</strong> {email} <br />
            </div>
          </div>
          <div className="extra content">
            <Link
              as="button"
              to={`/contact/${contact._id}`}
              className="ui primary icon button"
            >
              Ver visitas <i className="arrow right icon" />
            </Link>
          </div>
        </div>
      );
    });
  }

  render() {
    const { data, isFetching } = this.state;

    if (!isFetching && data.length) {
      return <div className="ui two cards">{this.renderContacts()}</div>;
    } else if (isFetching) {
      return <LoadingMsg />;
    } else {
      return (
        <div className="ui icon message">
          <i className="frown icon" />
          <div className="content">
            <div className="header">Nenhum contato adicionado até agora</div>
            <p>
              Talvez você não tenha configurado corretamente o script de
              rastreamento em seu site!
            </p>
          </div>
        </div>
      );
    }
  }
}

export default ContactList;
