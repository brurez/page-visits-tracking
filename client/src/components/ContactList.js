import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

import LoadingMsg from './LoadingMsg';
import axios from 'axios/index';
import socket from '../socket';

class ContactList extends Component {
  constructor(props){
    super(props);
    this.state = { data: [], isFetching: true };
    this.fetchApi = this.fetchApi.bind(this)
  }

  componentDidMount() {
    this.fetchApi();
    socket.on('new-contact', this.fetchApi);
  }

  fetchApi() {
    axios
      .get('/api/contacts')
      .then(res => {
        const { data } = res.data;
        this.setState({ data, isFetching: false });
      })
      .catch(err => {
        console.log(err);
        this.fetchApi();
      });
  }

  componentWillUnmount(){
    socket.off('new-contact', this.fetchApi);
  }

  renderContacts() {
    const { data } = this.state;

    return data.map(contact => {
      const { name, email, createdAt } = contact;

      return (
        <div className="card" key={contact._id}>
          <div className="content">
            <div className="header">{name}</div>
            <div className="meta">
              Criado em: {moment(createdAt).format('LLL')}
            </div>
            <div className="description">
              <strong>Email:</strong> {email} <br />
            </div>
          </div>
          <div className="extra content">
            <Link
              as="button"
              to={`/contact/${contact._id}`}
              className="ui primary icon button"
            >
              Ver visitas <i className="arrow right icon" />
            </Link>
          </div>
        </div>
      );
    });
  }

  render() {
    const { data, isFetching } = this.state;

    if (!isFetching && data.length) {
      return <div className="ui two stackable cards">{this.renderContacts()}</div>;
    } else if (isFetching) {
      return <LoadingMsg />;
    } else {
      return (
        <div className="ui icon message">
          <i className="frown icon" />
          <div className="content">
            <div className="header">Nenhum contato adicionado até agora</div>
            <p>
              Talvez você não tenha configurado corretamente o script de
              rastreamento em seu site!
            </p>
          </div>
        </div>
      );
    }
  }
}

export default ContactList;
