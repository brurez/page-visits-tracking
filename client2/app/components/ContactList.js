import React from 'react';

const ContactList = () => {
  return (
    <div>
      afafdasd
    </div>
  )
};

export default ContactList;
