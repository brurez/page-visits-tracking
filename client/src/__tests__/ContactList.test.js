import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import ContactList from '../components/ContactList';

jest.mock('axios');

configure({ adapter: new Adapter() });

describe('ContactList Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ContactList />);
  });

  test('if exist', () => {
    //console.log(wrapper);
    expect(wrapper).toBeTruthy();
  });

  test('if loading message is been displayed', () => {
    expect(wrapper.html()).toContain('Espere um segundo');
  });

  //after update

  test('if names are displayed after update', () => {
    wrapper.update();
    expect(wrapper.html()).toContain('Bruno de Rezende');
    expect(wrapper.html()).toContain('JoÃ£o do Teste');
  });

  test('if match the snapshot', () => {
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

