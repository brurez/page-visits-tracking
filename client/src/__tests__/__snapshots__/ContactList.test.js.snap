// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList Component if match the snapshot 1`] = `<Component />`;
